node/src/chain/mod.rs

```diff
@@ -170,23 +170,28 @@ impl ChainSrv {
         // Reset Consensus
         self.upper.abort();
 
+        // TODO: Ensure block is valid
+        // TODO: Call ExecuteStateTransition
+
         // Persist block
         db.read().await.update(|t| t.store_block(blk, true))?;
         self.most_recent_block = blk.clone();
 
+        // Delete from mempool any transaction already included in the block
```

Comment:
Not only, we should delete from the mempool any transaction using a nullifier the is included in the block.

Just to keep track, bot blocking the merge

Reply:
That's not how it is implemented in dusk-blockchain. 

In order to keep both repositories compatible with each other, I'd suggest to file separate issues - one in rusk repo and one in dusk-blockchain so that I address this suggestion.